docs/npm.md

```diff
@@ -0,0 +1,459 @@
+    },
+    "node_modules/fecha": {
+      "version": "4.2.3",
+      "resolved": "file:fecha-4.2.3.tgz",
+      "integrity": "sha512-OP2IUU6HeYKJi3i0z4A19kHMQoLVs4Hc+DPqqxI2h/DPZHTm/vjsfC6P0b4jCMy14XizLBqvndQ+UilD7707Jw==",
+      "license": "MIT"
+    },
+    "node_modules/foo": {
+      "resolved": "foo",
+      "link": true
+    },
+    "node_modules/is-positive": {
+      "version": "3.1.0",
+      "resolved": "git+ssh://git@github.com/kevva/is-positive.git#97edff6f525f192a3f83cea1944765f769ae2678",
+      "license": "MIT",
+      "engines": {
+        "node": ">=0.10.0"
+      }
+    }
+  }
+}
+
+```
+</details>
+
+<details>
+<summary>foo/package.json (workspace)</summary>
+
+```jsonc
+{
+  "name": "foo",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "devDependencies": {},
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+      "is-positive": "github:kevva/is-positive"
+  }
+}
+```
+</details>
+
+
+## Using fetched dependencies
+
+See also [usage.md](usage.md) for a complete example of Cachi2 usage.
+
+Cachi2 downloads the npm dependencies as tar archives into the `deps/npm/` subpath of the output directory.
+
+1. Dependencies fetched from npm registries are placed directly to this directory (array-flatten in the following example).
+1. Dependencies downloaded from other HTTPS URL are placed to subdirectory `external-<tarball_name>` (cachi2-bar in the following example).
+1. Dependencies retrieved from Git repository are placed to `host, namespace, repo` subdirectories (cachi2-foo in the following example).
+
+```text
+cachi2-output/deps/npm
+├── array-flatten-1.1.1.tgz
+├── bitbucket.org
+│        └── foo-testing
+│             └── cachi2-foo
+│                       └── cachi2-foo-external-gitcommit-9e164b97043a2d91bbeb992f6cc68a3d1015086a.tgz
+├── body-parser-1.20.1.tgz
+├── bytes-3.1.2.tgz
+│   ...
+├── external-cachi2-bar
+│        └── cachi2-bar-external-sha512-43e71f90ad5YOLO.tgz
+│   ...
+```
+
+Cachi2 updates [required files](#required-files) for offline installation in the hermetic build.
+These updates happen **automatically** by Cachi2's `inject-file` command.
+The rest of this section explains in detail which changes will be applied to the source repository.
+
+Update root package.json file together with package.json files for each [workspace](https://docs.npmjs.com/cli/v9/using-npm/workspaces?v=true) with changes:
+* For git repositories and HTTPS URLs in dependencies update their value to an empty string
+
+Update the following in the package-lock.json file:
+* Delete integrity items in packages requested as git repositories. Cachi2 will clone the repository and will create a tar archive with its content. Integrity checksum will change and will no longer match the previous integrity checksum, which would result in error.
```

Comment:
Older versions such as  `< npm v14`? If that is the case, I'd probably drop this behavior altogether (file a ticket for it)

Reply:
According to https://github.com/npm/cli/issues/2846#issuecomment-1153981036, npm 8.12.1 still did it. npm 9 does not AFAIK

Reply:
Since it was spotted in `npm 8.12.1` I believe this workaround must remain in the code. I changed explanation here, so it does not sound like Cachi2 does something insecure.

Reply:
Sounds good :+1: 

Could we just move the git integrity corner case below the two other points?

Reply:
Done!